src/SqsConsumer.ts

```diff
@@ -0,0 +1,181 @@
+   * if processing was not successful, then add the message to DLQ and delete
+   * the message from actual queue, to prevent queue flooding
+   */
+  async pollMessage() {
+    const params = {
+      AttributeNames: ['SentTimestamp'],
+      MaxNumberOfMessages: config.aws.sqs.sharedSnowplowQueue.maxMessages,
+      MessageAttributeNames: ['All'],
+      QueueUrl: config.aws.sqs.sharedSnowplowQueue.url,
+      VisibilityTimeout: config.aws.sqs.sharedSnowplowQueue.visibilityTimeout,
+      WaitTimeSeconds: config.aws.sqs.sharedSnowplowQueue.waitTimeSeconds,
+    };
+
+    let data: ReceiveMessageCommandOutput;
+    let eventBridgeContent: any; //body is generic based on event payload
+
+    try {
+      data = await this.sqsClient.send(new ReceiveMessageCommand(params));
+      if (data.Messages && data.Messages.length > 0) {
+        //body contains SNS payload
+        const body = JSON.parse(data.Messages[0].Body);
+        //get the eventBridge content from SNS.Message
+        eventBridgeContent = JSON.parse(body.Message);
+        console.log(`SQS body -> ` + JSON.stringify(body));
+      }
+    } catch (error) {
+      const receiveError = `Error receiving messages from queue ${JSON.stringify(
+        data?.Messages[0].Body
+      )}`;
+      console.error(receiveError, error);
+      Sentry.addBreadcrumb({ message: receiveError });
+      Sentry.captureException(error, { level: Sentry.Severity.Critical });
+    }
+    // Process any messages received and schedule next poll
+    if (eventBridgeContent != null) {
+      const status = await this.processMessage(eventBridgeContent);
+
+      if (!status) {
+        console.log(`adding to DLQ -> ${JSON.stringify(data.Messages[0])}`);
+        await this.insertToDLQ(data.Messages[0]);
+      }
+
+      //delete all messages as they are moved to DLQ
+      await this.deleteMessage(data.Messages[0]);
+
+      // Schedule next message poll
+      await this.scheduleNextPoll(
+        config.aws.sqs.sharedSnowplowQueue.afterMessagePollIntervalSeconds *
+          1000
+      );
+    } else {
+      // If no messages were found, schedule another poll after 5 mins
+      await this.scheduleNextPoll(
+        config.aws.sqs.sharedSnowplowQueue.defaultPollIntervalSeconds * 1000
+      );
+    }
+  }
+
+  /**
+   * returns false if it cannot process the given message
+   * any error is logged in sentry
+   * @param event, event bridge content from SNS payload
+   * @return true if processed successfully, false if error occurs
+   */
+  async processMessage(event: any) {
+    try {
+      const detailType = event['detail-type'];
+
+      if (eventConsumer[detailType] == null) {
+        throw new Error(
+          `Unable to retrieve handler for detailType='${detailType}'`
+        );
+      }
+
+      await eventConsumer[detailType](event);
+      return true;
+    } catch (error) {
+      const errorMessage = 'Error deleting message from queue';
```

Comment:
```suggestion
      const errorMessage = 'Error processing message from queue';
```

Error processing message from queue? Since deletion occurs elsewhere?